### PR TITLE
Closes #433: Updating lingering `rlang::abort()` calls

### LINKED ISSUE #433: Check for other calls to `rlang::abort()` and update with cli

### PR DESCRIPTION
**Pull request by ddsjoberg**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/dev/CONTRIBUTING.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/process_set_values_to.R [hunk 1]
```diff
@@ -70,20 +70,14 @@ process_set_values_to <- function(dataset,
...
-            " ",
-            names(set_values_to),
-            "=",
-            set_values_to,
-            collapse = "\n"
-          ),
-          "\n)\nError message:\n  ",
-          cnd
-        )
+      cli_abort(
```
Comment (bundfussr) [inline R/process_set_values_to.R:73]: Should we specify the `class` argument and add the `call` argument to `process_set_values_to()` and pass it to `cli_abort()`?
The error message
```
#> Error in value[[3L]](cond) : Assigning variables failed!
```
looks a bit strange.

File: R/assertions.R [hunk 2]
```diff
@@ -1694,7 +1694,7 @@ assert_one_to_one <- function(dataset,
         "i" = "Call {.fun get_many_to_one_dataset} to get all many-to-one values."
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply R/assertions.R]: Ooo, that is a fun update!

File: R/process_set_values_to.R (hunk 1 above)
Comment (ddsjoberg) [reply R/process_set_values_to.R:73]: Good catch. Updated!

image

**Review by bms63 — APPROVED** [review]

**Comment by bms63** [comment]
@bundfussr it looks like your comment was addressed.  Is it okay to merge this in?

**Comment by bundfussr** [comment]
> @bundfussr it looks like your comment was addressed. Is it okay to merge this in?

Yes, it is not perfect but we can merge it.

We still refer to functions the users don't know. For example:
```
> derive_extreme_records(data, dataset_add = data, order = exprs(AVAL), mode = "last", set_values_to = exprs(MAX = AVALC))
Error in `process_set_values_to()`:
! Assigning variables failed!
• `set_values_to = exprs(MAX = AVALC)`
See error message below:
ℹ In argument: `MAX = AVALC`. Caused by error: ! object 'AVALC' not found
```
However, we can't solve this issue in admiraldev. I am even not sure if we need to solve it. In principle it should be
```
> derive_extreme_records(data, dataset_add = data, order = exprs(AVAL), mode = "last", set_values_to = exprs(MAX = AVALC))
Error in `derive_extreme_records()`:
! Assigning variables failed!
• `set_values_to = exprs(MAX = AVALC)`
See error message below:
ℹ In argument: `MAX = AVALC`. Caused by error: ! object 'AVALC' not found
```
But most likely this would require a lot of updates in admiral and the information that the error occurred in `derive_extreme_records()` is not that helpful for users.
